Update Go module path to antrea.io/antrea/v2

sshivanshg opened the pull request:
Update the module path in go.mod and all imports from antrea.io/antrea to antrea.io/antrea/v2 to match the v2.x release series.

This resolves #6774 and ensures downstream consumers can correctly import the module.

antoninbas commented:
@jianjuns @tnqn @luolanzone what do you think? Are you ok with that change?

It matches the first step of my proposal described in https://github.com/antrea-io/antrea/issues/6774#issuecomment-2484417469

sshivanshg commented:
Hi @antoninbas,
I have now completed Step 2 of the module refactor:

- Added sub-modules for APIs (antrea.io/antrea/apis) and clientsets (antrea.io/antrea/client)
- Updated the main module and client code to use the new APIs sub-module for all API types
- Ensured type compatibility and tested all modules (go build and go test pass)
- Cleaned up the branch history to remove any large files

The branch is ready for your review. Please let me know if you have any questions or need further changes!

antoninbas commented:
@sshivanshg Please do not handle both steps in the same PR.
There is also something wrong with your current PR, it shows 3000+ commits.
